editor/src/components/canvas/canvas-strategies/flex-gap-strategy.ts

```diff
@@ -0,0 +1,184 @@
+import React from 'react'
+import { MetadataUtils } from '../../../core/model/element-metadata-utils'
+import { foldEither, isRight } from '../../../core/shared/either'
+import * as EP from '../../../core/shared/element-path'
+import {
+  ElementInstanceMetadata,
+  emptyComments,
+  isJSXElement,
+  jsxAttributeValue,
+  JSXElement,
+} from '../../../core/shared/element-template'
+import {
+  getJSXAttributeAtPath,
+  jsxSimpleAttributeToValue,
+  setJSXValuesAtPaths,
+  ValueAtPath,
+} from '../../../core/shared/jsx-attributes'
+import {
+  canvasPoint,
+  CanvasPoint,
+  CanvasRectangle,
+  magnitude,
+  rectContainsPoint,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import * as PP from '../../../core/shared/property-path'
+import {
+  EditorState,
+  forUnderlyingTargetFromEditorState,
+  modifyUnderlyingForOpenFile,
+  TransientFilesState,
+  withUnderlyingTargetFromEditorState,
+} from '../../editor/store/editor-state'
+import {
+  CanvasStrategy,
+  CanvasStrategyUpdateFnResult,
+  FlexAlignControlRectProps,
+  SelectModeCanvasSession,
+  SelectModeCanvasSessionProps,
+  SelectModeCanvasSessionState,
+} from './canvas-strategy-types'
+import { aperture, mapDropNulls } from '../../../core/shared/array-utils'
+import { stylePropPathMappingFn } from '../../inspector/common/property-path-hooks'
+import { optionalMap } from '../../../core/shared/optional-utils'
+
+export const flexGapStrategy: CanvasStrategy = {
+  name: 'Change Flex Gap',
+  fitnessFn: (editor, currentSession) => {
```

Comment:
Looks like the fitness functions need to check which control is actually being interacted with

Reply:
that's part of `currentSession`, so the signature is fine... I'm gonna put the check in the fitnessFn body. 
BUT in general, the two example strategies we have are more for demonstration and they are not intended to be production quality code

Reply:
fixed